synapse/storage/databases/main/signatures.py

```diff
@@ -41,18 +44,27 @@ async def get_event_reference_hashes(
 
         Returns:
              A mapping of event ID to a mapping of algorithm to hash.
+             Returns an empty dict for a given event id if that event is unknown.
```

Comment:
I should have said: it's not obvious to me that this is sensible behaviour, but it's what it was doing before, so I'm at least not making it worse.

(And somehow the sytests do hit this path.)